src/renderer/main.js

```diff
@@ -27,26 +27,39 @@ Vue.component('FontAwesomeIcon', FontAwesomeIcon)
 Vue.use(VueI18n)
 
 // List of locales approved for use
-const activeLocales = ['en-US', 'en_GB', 'ar', 'bg', 'cs', 'da', 'de-DE', 'el', 'es', 'es-MX', 'fi', 'fr-FR', 'gl', 'he', 'hu', 'hr', 'id', 'is', 'it', 'ja', 'nb_NO', 'nl', 'nn', 'pl', 'pt', 'pt-BR', 'pt-PT', 'ru', 'sk', 'sl', 'sv', 'tr', 'uk', 'vi', 'zh-CN', 'zh-TW']
+const activeLocales = ['system', 'en-US', 'en_GB', 'ar', 'bg', 'cs', 'da', 'de-DE', 'el', 'es', 'es-MX', 'fi', 'fr-FR', 'gl', 'he', 'hu', 'hr', 'id', 'is', 'it', 'ja', 'nb_NO', 'nl', 'nn', 'pl', 'pt', 'pt-BR', 'pt-PT', 'ru', 'sk', 'sl', 'sv', 'tr', 'uk', 'vi', 'zh-CN', 'zh-TW']
 const messages = {}
 /* eslint-disable-next-line */
 const fileLocation = isDev ? 'static/locales/' : `${__dirname}/static/locales/`
 
 // Take active locales and load respective YAML file
 activeLocales.forEach((locale) => {
+  // Import elctrons app object to access getLocale function
+  const { remote } = require('electron')
+  const { app } = remote
   try {
     // File location when running in dev
-    const doc = yaml.load(fs.readFileSync(`${fileLocation}${locale}.yaml`))
-    messages[locale] = doc
+    if (locale === 'system') {
+      // trying to find the corresponding .yaml file to systems locale
+      try {
+        const doc = yaml.load(fs.readFileSync(`${fileLocation}${app.getLocale()}.yaml`))
+        messages[locale] = doc
+      } catch (e) {
+        console.log(e)
+      }
+    } else {
+      const doc = yaml.load(fs.readFileSync(`${fileLocation}${locale}.yaml`))
+      messages[locale] = doc
+    }
```

Comment:
The aforementioned issue of having certain "missing" locales can be somewhat mended here.
You can first check if the locale exists.
If it doesn't, you can then check through the `activeLocales` to check if there's any locale that simply starts with those characters and, if there is, choose that one (or, if multiple exist, the first one).

Example:
if the app has locale `en`, you'll verify that there is no `en.yaml` (if there was, you'd load that one)
you then go through the locales and see that both `en-US` and `en_GB` locales exist (both start with `en`)
`en-US` occurs first, so you choose to load `en-US.yaml`

Reply:
So i created a Regular Expression 
```javascript
const regex = new RegExp(`${systemsLocale}-\\w+`, 'g')
```
So if systemsLocale is `fr` it matches with `fr-FR`

80e8466